compair/static/modules/answer/answer-module.js

```diff
@@ -242,61 +242,74 @@ module.controller(
+                        $scope.answer.file_id = file.id
+                    } else if ($scope.answer.file) {
+                        $scope.answer.file_id = $scope.answer.file.id;
                     } else {
-                        if (wasDraft) {
-                            Toaster.success("Answer Submitted");
-                        }
-                        else {
-                            Toaster.success("Answer Updated");
-                        }
-                        $location.path('/course/' + $scope.courseId + '/assignment/' + $scope.assignmentId);
-                        $route.reload();
+                        $scope.answer.file_id = null;
+                    }
 
-                        if (!$scope.canManageAssignment) {
-                            $location.search({'tab':'your_feedback', 'highlightAnswer':'1'});
+                    // copy answer so we don't lose draft state on failed submit
+                    var answer = angular.copy($scope.answer);
+                    answer.draft = !!saveDraft;
+                    answer.tracking = $scope.tracking.toParams();
+
+                    AnswerResource.save({'courseId': $scope.courseId, 'assignmentId': $scope.assignmentId}, answer).$promise.then(
+                        function (ret) {
+                            $scope.answer = ret;
+                            $scope.preventExit = false; // user has saved answer, does not need warning when leaving page
+                            $scope.refreshOnExit = ret.draft; // if draft was saved, we need to refresh to show updated data when modal is closed
+
+                            if (ret.draft) {
+                                Toaster.success("Answer Draft Saved", "Remember to submit your answer before the deadline.");
+                            } else {
+                                if (wasDraft) {
+                                    Toaster.success("Answer Submitted");
+                                }
+                                else {
+                                    Toaster.success("Answer Updated");
+                                }
+                                $location.path('/course/' + $scope.courseId + '/assignment/' + $scope.assignmentId);
+                                $route.reload();
+
+                                if (!$scope.canManageAssignment) {
+                                    $location.search({'tab':'your_feedback', 'highlightAnswer':'1'});
+                                }
+                            }
                         }
+                    ).finally(function() {
+                        $scope.submitted = false;
+                    });
+                }
+                else {
+                    $scope.submitted = false;
+                    Toaster.warning("Answer Not Saved", "Please check the highlighted fields and try saving again.");
+                    var firstErrorField = jQuery('.form-group.has-error')[0];
```

Comment:
You can do this a bit more angularly with
```
angular.element("[name='" + answerForm.$name + "']").find('.ng-invalid:visible:first').scrollIntoView();
```
or
```
angular.element("[name='" + answerForm.$name + "']").find('.form-group.has-error:first').scrollIntoView();
```

Reply:
Actually I don't think this works at all... anyways I'd prefer if jquery wasn't used inside of angular components if possible but if it must, it must